File: src/main/java/com/qlik/demo/java/model/TrainDocument.java
package com.qlik.demo.java.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "trains")
public record TrainDocument(
        @Id String id,
        String destination,
        SeatOption seatOption
) {
}

File: src/main/kotlin/com/qlik/demo/kotlin/model/TrainDocument.kt
package com.qlik.demo.kotlin.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "trains")
data class TrainDocument(
    @Id val id: String,
    val destination: String,
    val seatOption: SeatOption
)
